### ISSUE TITLE
[RFC] - Update GitHub Issue template to use YAML

### ISSUE THREAD

**Issue by gi-man** [darktable-org/darktable#13499]
**Is your feature request related to a problem? 
The problem is constantly asking users to provide all the info for their Issues. 

**Describe the solution you'd like**
Leverage: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

**Alternatives**
Keep current .md.

**Additional context**

I can start creating one if there is desire to implement it.

A quick high level concept:

Darktable source - Select (From darktable.org, self compiled, other)
Field to describe other:

Field to Describe Issue:

Field to Describe steps to reproduce:

Field for Expected:

_Optional fields_ for screen shot, screen cast, backtrace, logs, etc

Darktable version:
Platform: Select (Linux, Windows, Mac)
OS:
_optional fields for the other ones_

**Comment by MStraeten** [comment]
> **Is your feature request related to a problem?
> The problem is constantly asking users to provide all the info for their Issues.

when clicking on "new issue" i can select between 4 types of issue reports and all of them contain a template i can use or need to actively ignore it ;)

So why care too much on those, who ignores the demands of the guys that should solve the reporters issues?

key question for me: Will it be easier to use for those reporting most valuable issues yet?
At least for those the current way to provide different types of information infos in a simple way needs to be kept. 
Especially for some complex issue it's more comfortable to mix up text, logs, screenshots ...

**Comment by wpferguson** [comment]
> **Is your feature request related to a problem?
The problem is constantly asking users to provide all the info for their Issues.

We can ask for it up front, so that we can start working on the issue, or we can spend a day or two asking for it piece by piece until we can start working on the issue (or run out of patience)

**Comment by gi-man** [comment]
I developed a template and shared it in the #darktable IRC to get some initial feedback. I've addressed them and I think it is ready for more comments. I think it is easier to see in the form style than the .yml therefore I'm linking to my fork before doing the PR for it.

***Note: The intent of the form is for more general users and not for the experienced developers/contributors. I think we should keep a .md and call it _Developer Observations_ to just give a blank page. Experienced developers/contributors know how to provide clear details.***

https://github.com/gi-man/darktable/blob/Issue_Template/.github/ISSUE_TEMPLATE/bug_report.yml

I spent a lot of time trying to have a balance between required fields vs optional. More feedback on this area would be nice. I also added a label (needs triage) to aid in the organization. It is easy to turn off if not desired. And since I'm the one suggesting it, I would volunteer to assist with that task.
